crates/chain/src/tx_data_traits.rs

```diff
@@ -100,24 +100,30 @@ impl Anchor for ConfirmationBlockTime {
     }
 }
 
-/// An [`Anchor`] that can be constructed from a given block, block height and transaction position
-/// within the block.
-pub trait AnchorFromBlockPosition: Anchor {
-    /// Construct the anchor from a given `block`, block height and `tx_pos` within the block.
-    fn from_block_position(block: &bitcoin::Block, block_id: BlockId, tx_pos: usize) -> Self;
+/// Set of parameters sufficient to construct any [`Anchor`].
```

Comment:
```suggestion
/// Set of parameters sufficient to construct an [`Anchor`].
```

We can't guarantee that this can construct any [`Anchor`], since anyone can implement [`Anchor`] on a new type and the type may not impl `From<TxPosInBlock>`.

Reply:
Done.